src/ParallelAlgorithms/LinearSolver/Gmres/ElementActions.hpp

```diff
@@ -147,13 +176,55 @@ struct PrepareStep {
       const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
       const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
       const ParallelComponent* const /*meta*/) noexcept {
-    // Nothing to do before applying the linear operator to the operand
+    prepare_preconditioning(box, std::bool_constant<Preconditioned>{});
     return {std::move(box)};
   }
+
+ private:
+  template <typename DbTagsList>
+  static void prepare_preconditioning(
+      db::DataBox<DbTagsList>& box,
+      std::true_type /* preconditioning_enabled */) noexcept {
```

Comment:
Can't you just do `if constexpr` instead of an overload?

Reply:
I was still in cpp14 mode when writing this code :) I’ll see if I can move to `if constexpr`. It’s so much better.